arch/i386/mcount-dynamic.c

```diff
@@ -0,0 +1,202 @@
+#include <string.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+/* This should be defined before #include "utils.h" */
+#define PR_FMT     "dynamic"
+#define PR_DOMAIN  DBG_DYNAMIC
+
+#include "libmcount/internal.h"
+#include "utils/utils.h"
+#include "utils/symbol.h"
+
+#define PAGE_SIZE  4096
+#define XRAY_SECT  "xray_instr_map"
+
+/* target instrumentation function it needs to call */
+extern void __fentry__(void);
+
+struct xray_instr_map {
+	unsigned long addr;
+	unsigned long entry;
+	unsigned long type;
+	unsigned long count;
+};
+
+struct arch_dynamic_info {
+	struct xray_instr_map *xrmap;
+	unsigned xrmap_count;
+};
+
+int mcount_setup_trampoline(struct mcount_dynamic_info *mdi)
+{
+	/*
+	 * in i386, it is not possible to refer to eip.
+	 * so need a little trick. 
+	 * 
+	 * call 0x5
```

Comment:
call 0 ?

Reply:
@namhyung because referring eip is not possible for i386 
made a little trick which to be used for trampoline..